### PR TITLE
Category filter

### PR DESCRIPTION
**Pull request by Oluwadamilareolusakin**
# Description
This PR does the following:
* Enhances the Redux store
* The Redux store now keeps information about the current books filter which is a simple string with the name of the category.
* Creates the CHANGE_FILTER action in actions/index.js
* Creates a filter.js reducer which modifies filter in the Redux store (It defaults to All).
* Implement the CategoryFilter component and container
* The component renders a select box with filter categories.
* The book categories  are merged with the All option in the select box. All is not a valid book category and isn't saved in the Redux store, although it is a filter option.
* Keeps the filter value in the Redux store
* Renders the CategoryFilter component in the BooksList component.
* Implements the handleFilterChange method which modifies the filter state in the Redux store. Pass it down as a prop to CategoryFilter.
* Implements a conditional method which decides which books should be shown in the table depending on the filter value from Redux store.

### CODE REVIEW

**Comment by Oluwadamilareolusakin** [comment]
All errors have been fixed locally

File: .stickler.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# add the linters you want stickler to use for this project
+linters:
+  eslint:
+    # indicate where is the config file for stylelint
+    config: './.eslintrc.json'
+    
+files:
+  ignore: 
+    - 'src/*'
```
[the remark this answers is not in the document]
Comment (Oluwadamilareolusakin) [reply .stickler.yml]: Hello @jofer86, I ran eslint locally and fixed all errors

**Comment by Oluwadamilareolusakin** [comment]
Please refer to this [PR](https://github.com/Oluwadamilareolusakin/bookstore-react-redux/pull/11) for the updated version of this branch with errors fixed

File: .stickler.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    - '.eslintrc.json'
```
[the remark this answers is not in the document]
Comment (shloch) [reply .stickler.yml]: linter escape again

**Comment by shloch** [comment]
ok..

**Pull request merged**